src/server/pps/server/api_server.go

```diff
@@ -2030,53 +2053,148 @@ func job(kubeClient *kube.Client, jobInfo *persist.JobInfo, jobShimImage string,
 		MountPath: "/pach-bin",
 	})
 
+	return &jobOptions{
+		labels:             labels,
+		parallelism:        parallelism,
+		userImage:          userImage,
+		jobShimImage:       jobShimImage,
+		jobImagePullPolicy: jobImagePullPolicy,
+		jobEnv:             jobEnv,
+		volumes:            volumes,
+		volumeMounts:       volumeMounts,
+	}, nil
+}
+
+func service(kubeClient *kube.Client, jobInfo *persist.JobInfo, jobShimImage string, jobImagePullPolicy string, internalPort int32, externalPort int32) (*api.ReplicationController, *api.Service, error) {
+
+	options, err := getJobOptions(kubeClient, jobInfo, jobShimImage, jobImagePullPolicy)
+	if err != nil {
+		return nil, nil, err
+	}
+	rc := &api.ReplicationController{
+		TypeMeta: unversioned.TypeMeta{
+			Kind:       "ReplicationController",
+			APIVersion: "v1",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name:   jobInfo.JobID,
+			Labels: options.labels,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Selector: options.labels,
+			Replicas: options.parallelism,
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Name:   jobInfo.JobID,
+					Labels: options.labels,
+				},
+				Spec: api.PodSpec{
```

Comment:
This is repeated in the `job()` helper ... I could break out the PodSpec definition into its own helper as well.